src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/RecordAudio/RecordAudio.test.tsx

```diff
@@ -0,0 +1,172 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { renderWithProviders } from 'shared/utils';
+import * as useMediaUploadHooks from 'shared/hooks/useMediaUpload/useMediaUpload';
+
+import { RecordAudio } from './RecordAudio';
+import * as audioRecorderHooks from './RecordAudio.hooks';
+import { recordAudioDataTestid } from './RecordAudio.const';
+
+const mockClearBlob = jest.fn();
+const mockOnUpload = jest.fn();
+const mockOnChange = jest.fn();
+const mockOnClose = jest.fn();
+
+jest.mock('modules/Builder/components/MLPlayer', () => ({
+  ...jest.requireActual('modules/Builder/components/MLPlayer'),
+  MLPlayer: ({ onRemove, 'data-testid': dataTestid }) => (
+    <div data-testid={dataTestid}>
+      <button data-testid={`${dataTestid}-remove`} onClick={onRemove}></button>
+    </div>
+  ),
+}));
+
+describe('RecordAudio', () => {
+  let originalCreateObjectURL;
+
+  beforeEach(() => {
+    originalCreateObjectURL = window.URL.createObjectURL;
+    window.URL.createObjectURL = () => 'https://test.com/mock-record';
+  });
+
+  afterEach(() => {
+    window.URL.createObjectURL = originalCreateObjectURL;
+  });
+
+  afterAll(() => {
+    jest.clearAllMocks();
+  });
+
+  test('render RecordAudio component, test remove record', async () => {
+    const useAudioRecorderSpy = jest.spyOn(audioRecorderHooks, 'useAudioRecorder');
+    const useMediaUploadSpy = jest.spyOn(useMediaUploadHooks, 'useMediaUpload');
+
+    useAudioRecorderSpy.mockImplementation(({ setFile }) => ({
+      startRecording: jest.fn(),
+      stopRecording: () => {
+        // workaround for event listener
+        setTimeout(() => {
+          setFile({});
+        });
+      },
+      togglePauseResume: jest.fn(),
+      clearBlob: mockClearBlob,
+      isRecording: false,
+      isPaused: false,
+      isStopped: false,
+      recordingTime: 0,
+    }));
+    useMediaUploadSpy.mockImplementation(({ callback }) => ({
+        executeMediaUpload: () => {
+          callback?.('mock-url');
+        },
+        isLoading: false,
+        error: {
+          data: 'Mock Error',
+        },
+        stopUpload: jest.fn,
+      }));
+
+    renderWithProviders(
+      <RecordAudio open onUpload={mockOnUpload} onChange={mockOnChange} onClose={mockOnClose} />,
+    );
+
+    const recordButton = screen.getByTestId(`${recordAudioDataTestid}-record`);
+    await userEvent.click(recordButton);
+
+    const stopButton = screen.getByTestId(`${recordAudioDataTestid}-stop`);
+    await userEvent.click(stopButton);
+
+    useAudioRecorderSpy.mockImplementation(() => ({
```

Comment:
Do we need two mock implementations useAudioRecorderSpy in this test?

Reply:
yes, we need 2 spyOn as these are 2 separate tests, and in these tests, we need to mock the audio recording result (which is precisely what spyOn does here)